### ISSUE TITLE
Add metric to show vm-storage can't start big(monthly) merge

### ISSUE THREAD

**Issue by freeseacher** [VictoriaMetrics/VictoriaMetrics#686]
**Is your feature request related to a problem? Please describe.**
I'm pretty frustrated that can't know wheather vm-storage has enought space to do big merge or not. 
By the way i have at about 300Gb on each node. But merge process does not start. 
To make it start i have to add at about 800Gb. 

**Describe the solution you'd like**
An metric with gauge that show that long (probably month) merge is not ran due to space

**Describe alternatives you've considered**
Just add much more space. That is ok but not ok in longrun. 

**Additional context**

**Comment by valyala** [comment]
VictoriaMetrics tries performing both small and big merges every couple of seconds as described at https://medium.com/@valyala/how-victoriametrics-makes-instant-snapshots-for-multi-terabyte-time-series-data-e1f3fb0e0282 . The majority of attempts for big merge end up with `nothing to merge` status. So the counter for these statuses has little sense. Big merge is started when the following conditions are met:

* If there are multiple parts with roughly similar sizes, which can be merged into big part.
* If there is enough free space for storing big part additionally to source parts. Note that source parts are deleted only after successful merge. This guarantees data persistence in the event of unclean shutdown (OOM, kill -9, hard reset) - in this case VictoriaMetrics removes incomplete parts on next startup.

Note that VictoriaMetrics continues searching for smaller parts to be merged if bigger parts cannot be merged due to disk space shortage.

I'd recommend just monitoring `vm_free_disk_space_bytes` metric and ensuring that it is bigger than 50% of storage space occupied by VictoriaMetrics data, i.e. to alert on something like the following:
```metricsql
sum(vm_free_disk_space_bytes) by (instance) < 0.5 * sum(vm_data_size_bytes) by (instance)
```

**Comment by valyala** [comment]
Let's add `vm_big_merge_need_free_disk_space` metric, which will be set to 1 if VictoriaMetrics cannot create big part from smaller parts because there is no enough free disk space.

**Comment by valyala** [comment]
FYI, added `vm_merge_need_free_disk_space` metric to VictoriaMetrics in the following commits:

* Single-node version: bec9b31b815583627b4e76bb9b0f62118081aa06
* Cluster version: 7c2e4e267a513360b1e0ffdfef1f5ecc9a11748c

The `vm_merge_need_free_disk_space` value shows the number of per-month partitions with deferred background merges because of lack of free space. I.e. this is the number of potential background merges, which can start after increasing free disk space.

**Comment by valyala** [comment]
The `vm_merge_need_free_disk_space` metric is available starting from release [v1.42.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing this feature request as done.

**Comment by valyala** [comment]
It has been appeared that the `vm_merge_need_free_disk_space` metric confuses VictoriaMetric users - see, for example,  https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4915 and https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1682#issuecomment-938608067 . This metric shows how many partitions could start big merge if they had more free disk space. But this doesn't help resolving any issues:
- It is OK to have unmerged big partitions even if there is enough free disk space to make the merge. There are high chances that big parts already contain almost optimally compressed data, which results in high query performance. So an additional merge of such big partitions requires an additional disk read/write IO to perform the merge, while not bringing additional benefits such as lower disk space usage or faster query performance.
- It is OK to have a few tens of big parts instead of a few big parts. While this increases the number of open file descriptors, but the total number of file descriptors remains low.

So the decision is to drop the `vm_merge_need_free_disk_space` metric in order to reduce the level of confusion for users.

**Comment by valyala** [comment]
FYI, VictoriaMetrics stops exposing `vm_merge_need_free_disk_space` metric starting from [v1.94.0](https://docs.victoriametrics.com/CHANGELOG.html).
